test/e2e/satokensigner_test.go

```diff
@@ -0,0 +1,30 @@
+package e2e
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	clientcorev1 "k8s.io/client-go/kubernetes/typed/core/v1"
+
+	configclient "github.com/openshift/client-go/config/clientset/versioned/typed/config/v1"
+	test "github.com/openshift/cluster-kube-controller-manager-operator/test/library"
+)
+
+func TestSATokenSignerControllerSyncCerts(t *testing.T) {
+	// initialize clients
+	kubeConfig, err := test.NewClientConfigForTest()
+	require.NoError(t, err)
+	kubeClient, err := clientcorev1.NewForConfig(kubeConfig)
+	require.NoError(t, err)
+	configClient, err := configclient.NewForConfig(kubeConfig)
+	require.NoError(t, err)
+
+	test.WaitForKubeControllerManagerClusterOperatorAvailableNotProgressingNotDegraded(t, configClient)
+
+	err = kubeClient.Secrets("openshift-kube-controller-manager").Delete("service-account-private-key", &metav1.DeleteOptions{})
+	require.NoError(t, err)
+
```

Comment:
do something here to ensure you go progressing before starting your wait.

Reply:
Yeah, I realized I didn't even check if the secret gets recreated.